### PR TITLE
fix(DB/Creature): Steamvault - Coilfang Oracle casts Frostshock twice

### PR DESCRIPTION
**Pull request by Carriola-wotlk**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/19843
- Closes https://github.com/chromiecraft/chromiecraft/issues/6491

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/2024_09_01_00.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17803;
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 17803) AND (`source_type` = 0) AND (`id` IN (0));
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17803, 0, 0, 0, 0, 0, 100, 2, 5000, 8000, 13000, 16000, 0, 0, 11, 22582, 32, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Coilfang Oracle - In Combat - Cast \'Frost Shock\' (Normal Dungeon)');
```
Comment (alecsci) [inline data/sql/updates/pending_db_world/2024_09_01_00.sql:5]: Why do we need to escape the Frost Shock spell ?

File: data/sql/updates/pending_db_world/2024_09_01_00.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17803;
+
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 17803) AND (`source_type` = 0) AND (`id` IN (0));
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `event_param6`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
```
Comment (alecsci) [inline data/sql/updates/pending_db_world/2024_09_01_00.sql:4]: @Carriola-wotlk I am very curious what was the issue ?
I still see the chance field at 100%.

Comment (Carriola-wotlk) [reply data/sql/updates/pending_db_world/2024_09_01_00.sql:4]: castFlag set to 32
immagine

However, I just realized that I only fixed it for Normal dungeons. I'll update the query to also include Heroic dungeons

**Comment by Carriola-wotlk** [comment]
pr updated

**Comment by alecsci** [comment]
✅ Tested and the NPC now casts the spell as described in the SMARTAI section https://wowgaming.altervista.org/aowow/?npc=17803 (13-16 sec) and no more double casts.

**Comment by alecsci** [comment]
@Carriola-wotlk one more observation, the affected rows are 4.
Is this ok ? I assume 2 for normal, 2 for heroic, but why there are 2 ?

**Comment by Carriola-wotlk** [comment]
the affected rows are 2. Keira always builds queries in the same way, deleting the row and re-inserting the row with the new data. It always follows this flow, whether it is an insertion of a new row or an update.


But for example I could also just write a easier query using only one UPDATE satement, but I'm lazy and just copy and paste query builded by Keira XD

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**